components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -311,6 +200,122 @@ public Response apisPost(APIDTO body, MessageContext messageContext) {
         return null;
     }
 
+    /**
+     *
+     * @param body APIDTO of the API
+     * @return API
+     * @throws APIManagementException
+     */
+    private API createAPIByDTO(APIDTO body) throws APIManagementException {
+        APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+        String username = RestApiUtil.getLoggedInUsername();
+        List<String> apiSecuritySchemes = body.getSecurityScheme();//todo check list vs string
+        if (!apiProvider.isClientCertificateBasedAuthenticationConfigured() && apiSecuritySchemes != null) {
+            for (String apiSecurityScheme : apiSecuritySchemes) {
+                if (apiSecurityScheme.contains(APIConstants.API_SECURITY_MUTUAL_SSL)) {
+                    RestApiUtil.handleBadRequest("Mutual SSL Based authentication is not supported in this server", log);
+                }
+            }
+        }
+        if (body.getAccessControlRoles() != null) {
+            String errorMessage = RestApiPublisherUtils.validateUserRoles(body.getAccessControlRoles());
+
+            if (!errorMessage.isEmpty()) {
+                RestApiUtil.handleBadRequest(errorMessage, log);
+            }
+        }
+        if (body.getAdditionalProperties() != null) {
+            String errorMessage = RestApiPublisherUtils
+                    .validateAdditionalProperties(body.getAdditionalProperties());
```

Comment:
Doesn't seem to exceed 120 libe break? Why not in a single line?

Reply:
120 exceeded from ';'